modules/telemetry/include/hephaestus/telemetry/metric_record.h

```diff
@@ -0,0 +1,51 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+#include <string_view>
+
+#include "hephaestus/serdes/json.h"
+#include "hephaestus/telemetry/metric_sink.h"
+
+namespace heph::telemetry {
+namespace internal {
+[[nodiscard]] auto
+jsonToValuesMap(std::string_view json) -> std::unordered_map<std::string, Metric::ValueType>;
```

Comment:
I know I am late to the discussion, but I am not suuper in favor of the whole `Metric` name, as it is very easily mistaken with metric vs imperial units. So when you just see it, you can mistaken it for sth that has to do with unit transformations etc...